pep-0012.rst

```diff
@@ -568,35 +568,43 @@ the numbers will always match.
 Images
 ------
 
-If your PEP contains a diagram, you may include it in the processed
-output using the "image" directive::
+If your PEP contains a diagram or other graphic, you may include it in the
+processed output using the ``image`` directive:
+
+.. code-block:: rst
 
     .. image:: diagram.png
 
-Any browser-friendly graphics format is possible: .png, .jpeg, .gif,
-.tiff, etc.
+Any browser-friendly graphics format is possible; PNG should be
+preferred for graphics, JPEG for photos and GIF for animations.
```

Comment:
"PNG should be preferred for graphics, JPEG for photos" where does this guidance come from? 

Equally, do we want to allow animated GIF images in a PEP?

Reply:
> "PNG should be preferred for graphics, JPEG for photos" where does this guidance come from?

I'ts right in the name of each format, based on the their defined purpose and fundamental to how their respective compression strategies were designed.

JPEG was developed for use with photographs (the "P" in JPEG), and its DCT-based compression handles complex scenes, smooth gradients and high degrees of color variability well, but performs poorly on graphics, resulting in visibly poor quality yet larger file size.

By contrast, PNG was explicitly created for graphics (the "G" in PNG), and uses a lossless DEFLATE + filtering compression scheme plus color paletting to achieve excellent compression ratios on images with vector-style art, fewer discrete colors and defined shapes. but hugely balloons file size on photographs and similar relative to even a visually-lossless JPEG.

> Equally, do we want to allow animated GIF images in a PEP?

I have no opinion; we can elide that if others do. I included it because it was mentioned in the original text (TIFF was elided, as it is very much not web-friendly and its use cases for this were better covered by JPEG and PNG).

Reply:
Thanks for the explanation. To an extent my challenge was do we really need to care about the differences between JPEG and PNG. It's rare enough though that we can take case by case if need be. 

A

Reply:
I don't think it hurts to spend a single line mentioning which format to use, when we're mentioning the formats anyway.